src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,92 @@
+package duke.tasks;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.parser.DukeParseException;
+import duke.parser.Parser;
+
+/**
+ * Represents a Task.
+ * This class is abstract because tasks should exist as a Todo, Deadline or Event.
+ */
+public abstract class Task {
+    private final String desc;
+    private boolean isMarked;
+
+    /**
+     * Creates a Task given its description.
+     *
+     * @param desc the description of the task
+     */
+    public Task(String desc) {
+        this.desc = desc;
+        isMarked = false;
+    }
+
+    /**
+     * Decodes a task encoded string into a task.
+     *
+     * @param encodedTask the string representing the encoded task
+     * @return the task retrieved after decoding
+     */
+    public static Task decode(String encodedTask) {
+        Pattern pattern = Pattern.compile("^(?<type>.)\\|(?<mark>.)\\|(?<taskString>.+)$");
```

Comment:
Perhaps you could shorten this a bit and abstract out the code as it crosses the 30 line threshold